src/test-routes/error-with-secrets.ts

```diff
@@ -0,0 +1,55 @@
+import { app, json } from "../api";
+
+app.category("vanilla", () => {
+  // Returns an error response with secrets and PII in the headers and body.
+  app.get("/secrets/error", "ErrorWithSecrets", (req) => {
+    return {
+      status: 403,
+      headers: {
+        // Following headers should be redacted.
+        "x-ms-pii": "true",
+
+        // Following headers should not be redacted.
+        "x-ms-request-id": "5e123516-834e-4222-9e80-353108d33357",
+        "x-ms-version": "2022-02-01",
+      },
+      body: json({
+        error: {
+          code: "Unauthorized",
+          message: "The user 'user@contoso.com' is unauthorized.",
+          details: [
+            {
+              code: "UnauthorizedSharedKey",
+              innererror: "Shared key 1c88a67921784300a462b2cb61da2339 is not permitted access.",
+            },
+          ],
+          token: "1c88a67921784300a462b2cb61da2339",
+        },
+        primaryKey: "1c88a67921784300a462b2cb61da2339",
+        connectionString: "Key1=1c88a67921784300a462b2cb61da2339",
+      }),
+    };
+  });
+
+  app.post("/secrets/:create", "RequestWithSecrets", (req) => {
```

Comment:
is this supposed to be a parameter? I see it is also like that in the swagger file.
express.js will assume you have  a path param called `create` here.

Reply:
It's an action, following Azure guidelines.

Reply:
`/secrets/:create` is valid: https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#performing-an-action, though, @JeffreyRichter @mikekistler I see that document wasn't updated to say that `/:` was allowed now that the Postman issue I raised has been resolved.

curl returns 303 if I try http://localhost:3000/secrets/notcreate so express seems to support it just fine as well.

Reply:
Sorry, I didn't set the write method on the attempt. Using POST "notcreate" does "work", but escaping with `[:]` (per online docs) works correctly.

Reply:
The Azure Guidelines are to use a ":" with no preceding "/". We discussed allowing "/:" and decided against it since we want to promote a single pattern wherever possible for consistency.

Reply:
Thanks. Based on https://github.com/microsoft/api-guidelines/issues/265 I see I misremembered. Do you feel this should be changed in this test service to reflect that, or not a big deal? Happy to submit a PR if it should be changed.

Reply:
Not a big deal for this to be in the test server -- I would not go to the trouble of changing it now.